src/Tasks/ResolveSDKReference.cs

```diff
@@ -1251,7 +1251,7 @@ private void CreateResolvedReferenceItem(string targetConfiguration, string targ
                     AddResolutionWarning("ResolveSDKReference.MaxPlatformVersionNotSpecified", projectName, DisplayName, Version, targetPlatformIdentifier, targetPlatformVersionFromItem.ToString(), targetPlatformIdentifier, targetPlatformVersion.ToString());
                 }
 
-                if (!String.IsNullOrEmpty(TargetPlatform) && !String.Equals(targetPlatformIdentifier, TargetPlatform))
+                if (!String.IsNullOrEmpty(TargetPlatform) && !String.Equals(targetPlatformIdentifier, TargetPlatform) && (!String.Equals("UAP", TargetPlatform, StringComparison.OrdinalIgnoreCase) || !String.Equals(targetPlatformIdentifier, "windows", StringComparison.OrdinalIgnoreCase)))
```

Comment:
```suggestion
                if (!String.IsNullOrEmpty(TargetPlatform) && !string.Equals(targetPlatformIdentifier, TargetPlatform) && (!String.Equals("UAP", TargetPlatform, StringComparison.OrdinalIgnoreCase) || !string.Equals(targetPlatformIdentifier, "windows", StringComparison.OrdinalIgnoreCase)))
```

nit

Reply:
I agree, but I did it this way intentionally because the rest of the file uses String, and I didn't want to change everything or stand out.